### ISSUE TITLE
External tilesets are disallowed inside multiple contents

### ISSUE THREAD

**Issue by javagl** [CesiumGS/cesium#11960]
The title basically says it all: When using multiple contents that refer to external tilesets, then a message
> Error: External tilesets are disallowed inside multiple contents

is printed and the external tilesets are not rendered.

This _might_ be a leftover from the `3DDILES_multiple_contents` extension: This explicitly disallowed external tilesets in the contents (see https://github.com/CesiumGS/3d-tiles/pull/583/files ). But in 3D Tiles 1.1, there is no such restriction.

Attached a _really_ minimal test case of one tileset that only has two contents, which in turn are (external) tilesets, to reproduce the issue:

[external tilesets in multiple contents.zip](https://github.com/CesiumGS/cesium/files/15134199/external.tilesets.in.multiple.contents.zip)

**Comment by javagl** [comment]
I did have a short look at this. And I pragmatically created a test case:

- a tileset with a root that as 4 `contents`
- the _first_ content is a *red* unit cube GLB
- the other contents are three external tilesets (that refer to the green, blue, and white content, respectively)

[multipleContentsExternal 2024-12-04b.zip](https://github.com/user-attachments/files/18014662/multipleContentsExternal.2024-12-04b.zip)

Then, I simply commented out the [check for external tilesets in multiple contents](https://github.com/CesiumGS/cesium/blob/382c517418bb78be96d5a1988085bf3b9bf1c467/packages/engine/Source/Scene/Multiple3DTileContent.js#L521), and it works:

![Cesium Multiple External 0001](https://github.com/user-attachments/assets/bcfd0f29-304f-4e55-b2a2-5c67e713ff20)

Yay 🎉 

But ... there is a flag `tile.hasTilesetContent`. This is a `boolean` that is set for tiles that have external tilesets as content. Now ... one could be tempted to think that the flag still can to `true` here, because after all, this tile **does** have tileset content. But it also has non-tileset content 🤔 

Searching for this flag reveals several places where this is checked. Among them are things like this...
```javascript
/**
 * @private
 * @param {Cesium3DTileset} tileset
 * @param {Cesium3DTile} tile
 * @returns {number}
 */
function getPriorityReverseScreenSpaceError(tileset, tile) {
  const parent = tile.parent;
  const useParentScreenSpaceError =
    defined(parent) &&
    (!tileset.isSkippingLevelOfDetail ||
      tile._screenSpaceError === 0.0 ||
      parent.hasTilesetContent ||
      parent.hasImplicitContent);
  const screenSpaceError = useParentScreenSpaceError
    ? parent._screenSpaceError
    : tile._screenSpaceError;
  return tileset.root._screenSpaceError - screenSpaceError;
}
```
Can I be sure that setting this flag to `true` will not cause a bug here? No. Similarly for all the other places where this flag is queried (although most of them at least _look_ like they might be easier to understand ... and interestingly, this flag is _always_ checked in `tile.hasTilesetContent || tile.hasImplicitContent` - never alone)

I can read more, and do more tests, but there's always the risk that I miss a case - like the one where the `_priorityReverseScreenSpaceError` is computed and used. 

So if someone can say from the tip of the head of whether it's safe to simply _allow_ external tilesets in multiple contents (and set `hasTilesetContent=true`), that could safe a lot of time...

**Comment by ggetz** [comment]
@javagl Good point, this could have implications on tileset refinement, similar to the issue that came up here: https://github.com/CesiumGS/cesium/issues/9356

I haven't thoroughly checked that the case you bring up here is safe to change, but I would start by looking through [Cesium3DTilesetTraversal.js](https://github.com/CesiumGS/cesium/blob/main/packages/engine/Source/Scene/Cesium3DTilesetTraversal.js) to confirm.

**Comment by javagl** [comment]
@ggetz That thoroughness is the main difficulty when trying to address this. It might be that a bug shows up when one tile has two children of which one is empty and the other has two contents where one is a GLB and the other is an external tileset that is an implicit tilest that has a geometric error that is smaller than the geometric error of the parent tile and 'skipLevelOfDetail' is set to true  ...?! 🥴 ?! The point is: The "toy test" that I created doesn't say much...  

The only approach that I can imagine (and that's still on my TODO list) is to search for all uses of `"hasTilesetContent"`, see how this flag affects that part of the code, and then try to understand the code to make somewhat educated guesses about possible implications (maybe by intentionally constructing test cases that "provoke" a difference between `hasTilesetContent` being `true/false`).

**Comment by javagl** [comment]
This should be fixed now via https://github.com/CesiumGS/cesium/pull/12440
